### ISSUE TITLE
name "Logger is not defined"

### ISSUE THREAD

**Issue by muzzi123** [diegoalejogm/gans#22]
![1](https://user-images.githubusercontent.com/50144512/95018920-c7962680-0662-11eb-92c5-664c3a12357d.JPG)
![3](https://user-images.githubusercontent.com/50144512/95018923-cb29ad80-0662-11eb-90fb-7974c79f46a6.JPG)

Hello Everyone!

 I have two problems when I run this whole script by giving my path of MNIST data

by data folder path is this "C:\Users\muzam\Anaconda3\envs\tensorflow\Data_Practice\train-images-idx3-ubyte\dataset\MNIST"

when I run the "Training code" in juypter

# Create logger instance
logger = Logger(model_name='VGAN', data_name='MNIST')
# Total number of epochs to train
num_epochs = 20
for epoch in range(num_epochs):
    for n_batch, (real_batch,_) in enumerate(data_loader):
        N = real_batch.size(0)
        # 1. Train Discriminator
        real_data = Variable(images_to_vectors(real_batch))
        # Generate fake data and detach 
        # (so gradients are not calculated for generator)
        fake_data = generator(noise(N)).detach()
        # Train D
        d_error, d_pred_real, d_pred_fake = \
              train_discriminator(d_optimizer, real_data, fake_data)

        # 2. Train Generator
        # Generate fake data
        fake_data = generator(noise(N))
        # Train G
        g_error = train_generator(g_optimizer, fake_data)
        # Log batch error
        logger.log(d_error, g_error, epoch, n_batch, num_batches)
        # Display Progress every few batches
        if (n_batch) % 100 == 0: 
            test_images = vectors_to_images(generator(test_noise))
            test_images = test_images.data
            logger.log_images(
                test_images, num_test_samples, 
                epoch, n_batch, num_batches
            );
            # Display status Logs
            logger.display_status(
                epoch, num_epochs, n_batch, num_batches,
                d_error, g_error, d_pred_real, d_pred_fake
            )

First it give the error that my kernel is dead 

![image](https://user-images.githubusercontent.com/50144512/95019055-87837380-0663-11eb-9621-2b2a182eb6ac.png)

then it after kernel is again idle when I run it again it give me the error "name Logger is not defined as shown below

![image](https://user-images.githubusercontent.com/50144512/95019074-b7327b80-0663-11eb-84ef-a6c2bafa3fc2.png)

And it also generate the folder in data in which subdirectories VGAN and MNIST but this folder are empty.

![image](https://user-images.githubusercontent.com/50144512/95019122-ed6ffb00-0663-11eb-847d-b446e59cacc3.png)

I don't know what the problem can any one tell me the solution.

Thank you very much for you time and consideration. I am waiting for your reply.

Regards:

Muhammad Muzammil Mukhtar

**Comment by diegoalejogm** [comment]
Hi Muhammad,

The second error occurs because the kernel restarts and you need to import `Logger` again with `import Logger`.
Could you please share the error message of the colab in this discussion? Does it mention why the Kernel died?

**Comment by diegoalejogm** [comment]
Closed the issue. Please re-open if you keep having the same problem
